library/public/types.h

```diff
@@ -58,6 +71,54 @@ class F3D_EXPORT ratio_t
   double Value;
 };
 
+/**
+ * Describe a RGB color.
+ */
+class F3D_EXPORT color_t
+{
+public:
+  color_t() = default;
+  color_t(const std::vector<double>& vec)
+  {
+    if (vec.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided vector does not have the right size");
+    }
+    std::copy_n(vec.begin(), 3, this->RGB.begin());
+  }
+  color_t(double red, double green, double blue)
+  {
+    this->RGB[0] = red;
+    this->RGB[1] = green;
+    this->RGB[2] = blue;
+  }
+  template<std::size_t N>
+  color_t(const double (&list)[N])
+  {
+    static_assert(N == 3, "color_t needs exactly 3 elements to initialize");
+    std::copy_n(list, 3, this->RGB.begin());
+  }
```

Comment:
Do we need a template here? Why not just:
```suggestion
  color_t(const double (&list)[3])
  {
    std::copy_n(list, 3, this->RGB.begin());
  }
```

Reply:
> then you loose the static_assert, which prevent misuse.

the `3` is hard-coded so it should also fail at compile time, no?

Reply:
`static_assert(sizeof(list) == 3 * sizeof(double), "color_t needs exactly 3 elements to initialize");` ?